src/common/SelectableOptionsScreen.js

```diff
@@ -0,0 +1,101 @@
+/* @flow strict-local */
+
+import React, { useMemo } from 'react';
+import type { Node } from 'react';
+import { LogBox, View } from 'react-native';
+
+import type { RouteProp } from '../react-navigation';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+import Screen from './Screen';
+import SelectableOptionRow from './SelectableOptionRow';
+import ZulipTextIntl from './ZulipTextIntl';
+import { createStyleSheet } from '../styles';
+
+type Item<TKey> = $ReadOnly<{|
+  key: TKey,
+  title: string,
+  subtitle?: string,
+  selected: boolean,
+|}>;
+
+type Props<TItemKey> = $ReadOnly<{|
+  navigation: AppNavigationProp<'selectable-options'>,
+  route: RouteProp<
+    'selectable-options',
+    {|
+      title: string,
+
+      /**
+       * An optional explanation, to be shown before the items.
+       */
+      description?: string,
+
+      items: $ReadOnlyArray<Item<TItemKey>>,
+
+      // This param is a function, so React Nav is right to point out that
+      // it isn't serializable. But this is fine as long as we don't try to
+      // persist navigation state for this screen or set up deep linking to
+      // it, hence the LogBox suppression below.
+      //
+      // React Navigation doesn't offer a more sensible way to have us pass
+      // the selection to the calling screen. …We could store the selection
+      // as a route param on the calling screen, or in Redux. But from this
+      // screen's perspective, that's basically just setting a global
+      // variable. Better to offer this explicit, side-effect-free way for
+      // the data to flow where it should, when it should.
+      onRequestSelectionChange: (itemKey: TItemKey, requestedValue: boolean) => void,
+    |},
+  >,
+|}>;
+
+// React Navigation would give us a console warning about non-serializable
+// route params. For more about the warning, see
+//   https://reactnavigation.org/docs/5.x/troubleshooting/#i-get-the-warning-non-serializable-values-were-found-in-the-navigation-state
+// See comment on this param, above.
+LogBox.ignoreLogs([/selectable-options > params\.onRequestSelectionChange \(Function\)/]);
```

Comment:
This all makes sense! Thanks for the explanation.

Reply:
It's basically copied from `EmojiPickerScreen`. 😛

Reply:
Ha, OK -- I thought it sounded a bit familiar, but figured I must have looked at a version of this PR earlier.

Still, that's also text that you wrote, so the thanks aren't wrong :slightly_smiling_face:

Reply:
The reasoning was yours, though, as I remember! We can share the thanks, then? 😅